.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,64 @@
+name: Go Test Workflow
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '^1.19'
+    
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+      
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+
+    - name: Fetch base branch
+      if: ${{ github.event_name == 'pull_request' }}
+      run: git fetch origin ${{ github.base_ref }}
+      
+    - name: Run tests
+      run: |
+        go test ./... -cover
+        
+    - name: Download golangci-lint
+      run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
+    
+    - name: golangci-lint
+      if: ${{ github.event_name == 'pull_request' }}
+      run: |
+        $(go env GOPATH)/bin/golangci-lint run --out-format checkstyle --issues-exit-code=0 --new-from-rev=origin/${{ github.base_ref }} ./... > golanglint.xml
+    
+    - name: golangci-lint-on-push
+      if: ${{ github.event_name == 'push' }}
+      run: |
+        $(go env GOPATH)/bin/golangci-lint run --out-format checkstyle --issues-exit-code=0 ./... > golanglint.xml
+        
+    - name: SonarCloud Scan
```

Comment:
This seems to have a problem. If you don't need SonarCloud, you don't have to use it. I really find the Github security tab with CodeQL and Secret scanning more useful.